--- src/src/corelib/plugin/qlibrary_unix.cpp	2020-10-27 08:02:11.000000000 +0000
+++ src.mod/src/corelib/plugin/qlibrary_unix.cpp	2021-05-04 15:57:25.924898438 +0000
@@ -52,7 +52,7 @@
 #  include <private/qcore_mac_p.h>
 #endif
 
-#ifdef Q_OS_ANDROID
+#if defined(Q_OS_ANDROID) && !defined(Q_OS_ANDROID_EMBEDDED)
 #  include <private/qjnihelpers_p.h>
 #endif
 
@@ -100,7 +100,7 @@
         suffixes << QLatin1String(".so.%1").arg(fullVersion);
     } else {
         suffixes << QLatin1String(".so");
-# ifdef Q_OS_ANDROID
+# if defined(Q_OS_ANDROID) && !defined(Q_OS_ANDROID_EMBEDDED)
         suffixes << QStringLiteral(LIBS_SUFFIX);
 # endif
     }
@@ -167,7 +167,7 @@
     // library is reloaded with dlopen() at a later time.
 #if defined(RTLD_NODELETE)
     if (loadHints & QLibrary::PreventUnloadHint) {
-#   ifdef Q_OS_ANDROID // RTLD_NODELETE flag is supported by Android 23+
+#   if defined(Q_OS_ANDROID) && !defined(Q_OS_ANDROID_EMBEDDED) // RTLD_NODELETE flag is supported by Android 23+
         if (QtAndroidPrivate::androidSdkVersion() > 22)
 #   endif
             dlFlags |= RTLD_NODELETE;
@@ -236,7 +236,7 @@
             }
 
             hnd = dlopen(QFile::encodeName(attempt), dlFlags);
-#ifdef Q_OS_ANDROID
+#if defined(Q_OS_ANDROID) && !defined(Q_OS_ANDROID_EMBEDDED)
             if (!hnd) {
                 auto attemptFromBundle = attempt;
                 hnd = dlopen(QFile::encodeName(attemptFromBundle.replace(QLatin1Char('/'), QLatin1Char('_'))), dlFlags);
